tests/requests/test_subscription_request.py

```diff
@@ -0,0 +1,108 @@
+from unittest import TestCase
+
+from chargebee_byte.requests.subscription_request import SubscriptionRequest
+
+
+class TestSubscriptionRequest(TestCase):
+    def setUp(self):
+        self.request = SubscriptionRequest()
+
+    def test_path_is_set_to_subscriptions(self):
+        self.assertEqual(self.request.path, '/subscriptions')
+
+    def test_has_correct_allowed_params_set(self):
+        self.assertCountEqual(self.request.allowed_parameters, [
```

Comment:
If we're going to maintain a list like this anyway there's no point in having the dynamic generation of parameters.

I think this entire test can be dropped, and just check that the `generate_whatever_params` functions work as expected. Just my 2 cents though.

Reply:
Well we already have tests for the `generate_*_params` functions, but I just want to be explicitly sure that all allowed parameters are present. Would you mind if I kept it in? Personally I see the benefit of the `generate_*_params` functions since it cleans up the code somewhat and makes it easy to re-use for other requests later on.